src/main/webapp/app/text-editor/text-editor.service.ts

```diff
@@ -1,12 +1,37 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
+import { TextSubmission } from 'app/entities/text-submission';
+import { Language } from 'app/entities/tutor-group';
+import { Franc, FrancLanguage, SUPPORTED_TEXT_LANGUAGES } from './franc';
 
 @Injectable({ providedIn: 'root' })
 export class TextEditorService {
+    readonly franc: Franc = require('franc-min');
+
     constructor(private http: HttpClient) {}
 
     get(id: number): Observable<any> {
         return this.http.get(`api/text-editor/${id}`, { responseType: 'json' });
     }
+
+    determineLanguage(submission: TextSubmission): void {
```

Comment:
It would be better if `determinLanguage` would just return the predicted language, so the callee can then do with the language whatever they want (in this case assign it to the submission).

A couple of arguments for this change:
- `determineLanguage` does not indicate by its name that it will mutate the given submission
- The reusability potential would be higher

Reply:
there is already a method for predicting language(`predictLanguage`), which is called by  `determineLanguage`. Would `setLanguage` be more suitable then `determineLanguage` ?

Reply:
I would personally do it directly in the component.
`this.submission.language = this.textEditorService.predictLanguage(submission.text);`

The advantage is that the textEditorService then doesn't mutate the submission, so when debugging it is easier to find places where the submission could have been changed.